src/esm_parser/esm_parser.py

```diff
@@ -1527,6 +1527,85 @@ def resolve_choose(model_with_choose, choose_key, config):
         logging.debug("key=%s", key)
 
 
+def resolve_early_var(
+    var, config, current_model=None, user_config={}, model_config={}, setup_config={}
+):
+
+    #import ipdb
+    if not current_model:
+        logger.error(
+            "Please define a ``current_model`` for ``resolve_early_var`` method"
+        )
+        sys.exit(1)
+    elif current_model in config:
+        this_config = config
+    elif current_model == config.get("model"):
+        this_config = {current_model: config}
+    else:
+        logger.error(f"``current_model: {current_model}`` and ``config`` do not correspond!")
+        return
```

Comment:
`return`, or do we want to exit?

```suggestion
        sys.exit(1)
```